Upgrade solvation thermo GAV and solute & solvent libraries

yunsiechung opened the pull request:
_**Note that this pull request requires simultaneous updates of RMG-Py and RMG-database. The pull request on RMG-Py solvation_thermo_GAV branch is https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2112. The last commit needs to be removed before merging this branch.**_

### Motivation or Problem
The previous group additivity (GAV) method for solvation thermo uses very limited number of groups. This pull request will upgrade the solvation thermo GAV method that uses a lot more groups. Also, this pull request adds a lot more solvents and solutes to the solvent and solute libraries. 

### Description of Changes
The major changes include:

1. The solvation GAV now follows an approach similar to the gas-phase thermo GAV and uses `group`, `ring`, `polycyclic`, `longDistanceInteraction_cyclic`, `longDistanceInteraction_noncyclic`, `halogen`, and `radical` corrections to estimate solute parameters (E, S, A, B, L). All the solvation groups are fitted by me (more detailed reference will be added to the RMG-database once this work get published). Each solvation group includes a new attribute `dataCount` stored as either a `DataCountGAV` class or `None`. This new attribute stores the number of experimental solute parameter data used to fit each group value.

2. Many more solutes and solvents are added to the solute and solvent libraries. The new solvent parameters are fitted by me (more detailed reference will be added to the RMG-database once this work get published), and the number of solvation free energy and enthalpy data used to fit these parameters are included in the solvent library entries. The solvation free energy and enthalpy errors associated with the fitted solvent parameters are also included.

mjohnson541 commented on input/thermo/libraries/primaryThermoLibrary.py:
```diff
@@ -1136,3 +1136,39 @@
+
+entry(
+    index = 46,
+    label = "OCCCO",
```
Should we have similar molecules such as  O=C=C=O as well in primaryThermoLibrary?

yunsiechung replied:
This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

mjohnson541 commented on input/thermo/libraries/primaryThermoLibrary.py:
```diff
@@ -903,13 +903,17 @@
         CpInf = (37.4151,'J/mol/K'),
     ),
-    shortDesc = u"""71STUPRO""",
+    shortDesc = u"""71STUPRO JANAF 2nd ed. (1971)""",
```
Do we have an idea how accurate these are?

yunsiechung replied:
This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

mjohnson541 commented on input/solvation/libraries/solute.py:
```diff
@@ -4,8 +4,9 @@
 shortDesc = u""
 longDesc = u"""
-From Abraham J. Chem. Soc. 1994
+From Abraham J. Chem. Soc. 1994 or from in-house database from Prof. Abraham
```
In both of these libraries (this commit and the one before, and for the groups files as well (although those are likely to be more homogeneous)) there are lots of entries without long descriptions. Can you say something about the default source of values at the top of the file and add a description for the other ones?

yunsiechung replied:
I added the reference in the long description of each entry for `solvent.py` and `solute.py` (solvent and solute libraries), which are done in the fixup commits 4fa1bb3 and 4b83d36.
I added a bit more detailed explanation in the long description at the top of the file for solvation group files in the fixup commit ebdaf4c.

mjohnson541 reviewed (commented):
Thanks! This needs to pass unit tests, but the commits look mostly okay. Lots of RMG-tests changes in the solvation example do the changes look reasonable to you?

yunsiechung commented on input/thermo/libraries/primaryThermoLibrary.py:
```diff
@@ -884,3 +884,255 @@
+        Cp0 = (29.1007,'J/mol/K'),
+        CpInf = (37.4151,'J/mol/K'),
+    ),
+    shortDesc = u"""71STUPRO""",
```
This is not a commit added by me, so I will close this. I think the previous version was not rebased properly with the official/master branch and somehow this PR showed other people's commits as well. I rebased again and pushed again.

yunsiechung commented:
@mjohnson541 I made all the changes you requested. You can see my comment for each of your requested change.

If everything looks good for both this and its twin RMG-Py PR  [#2112](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2112), I can squash the fixup commits, remove the temporary commit for simultaneous updates of RMG-Py and RMG-database (30f35ac), and force push.

mjohnson541 reviewed (approved):
Looks good to squash and then merge to me!

yunsiechung commented:
@mjohnson541 I squashed and rebased. But I did not remove the temporary commit (0155ed3) just to make sure this passes all the tests.
When you merge, please remove the temporary commit (0155ed3) for both this and RMG-Py PR.
